Formula/idevicerestore.rb

```diff
@@ -0,0 +1,34 @@
+class Idevicerestore < Formula
+  desc "Tools to restore and recover iDevices"
+  homepage "https://www.libimobiledevice.org/"
+
+  head do
+    url "https://git.libimobiledevice.org/idevicerestore.git"
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+    depends_on "libxml2"
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "libimobiledevice"
+  depends_on "libirecovery"
+  depends_on "libplist"
+  depends_on "libtasn1"
+  depends_on "libusbmuxd"
+  depends_on "libzip"
+  depends_on "openssl@1.1"
+
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+
+    system "make", "install"
+  end
+
+  test do
+    system "idevicerestore", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is another case where a physical device would have to be present to do any useful testing...